### PR TITLE
same change find with email

### PR DESCRIPTION
**Pull request by sirinebettaieb**

### CODE REVIEW

**Review by Fatimata-dev — APPROVED** [review]

File: src/main/java/com/leveluptasks/service/GroupService.java [hunk 1]
```diff
@@ -37,50 +41,40 @@ public Groupe getGroupById(Long id) {
...
+            } else {
+                throw new IllegalArgumentException("User not found with email: " + userEmail);
             }
         }
-        groupRepository.save(groupe);
-        return result;
 
-      
+        groupRepository.save(groupe);
+        return "OK";
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/service/GroupService.java]: better return the group itself if you want to display it, or at least a response entity. Beter than a simple String

File: src/main/java/com/leveluptasks/service/GroupService.java [hunk 2]
```diff
@@ -37,50 +41,40 @@ public Groupe getGroupById(Long id) {
...
         groupRepository.deleteById(id);
     }
 
-    public String updateGroup(Long id, GroupCreationDTO groupCreationDTO) {
-        Groupe existingGroup = groupRepository.findById(id).orElseThrow(() -> new IllegalArgumentException("Groupe avec l'ID '" + id + "' non trouvé")); 
-        existingGroup.setName(groupCreationDTO.getGroupName());
-        groupRepository.save(existingGroup);
-        return "OK";
+    public Groupe  updateGroup( GroupCreationDTO groupCreationDTO) {
+        return groupRepository.save(new Groupe(groupCreationDTO.getGroupName()));
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/service/GroupService.java]: well, am not sure you are updating the groupe doing this

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/leveluptasks/controller/GroupController.java [hunk 3]
```diff
@@ -37,15 +37,21 @@ public Groupe getGroupById(@PathVariable Long id) {
...
+   
     @PutMapping("/{id}")
-    public Groupe updateGroup( @RequestBody GroupCreationDTO groupCreationDTO, @PathVariable Long id) throws NoSuchAlgorithmException {
-        return groupService.updateGroup( groupCreationDTO);
+    public Groupe updateGroupe( @RequestBody Groupe groupe, @PathVariable Long id) throws NoSuchAlgorithmException {
+        return groupService.updateGroupe( id, groupe);
     }
 
-   
+     @PutMapping("/{id}/addUser")
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/controller/GroupController.java:48]: postmapping if you add a new user

File: src/main/java/com/leveluptasks/controller/GroupController.java [hunk 4]
```diff
@@ -37,15 +37,21 @@ public Groupe getGroupById(@PathVariable Long id) {
...
+    public Groupe addUseGroupe( @RequestBody List<String> UserEmail, @PathVariable Long id) throws NoSuchAlgorithmException {
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/controller/GroupController.java]: addUserToGroup

File: src/main/java/com/leveluptasks/dto/GroupCreationDTO.java [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+import java.util.List;
+
 import jakarta.validation.constraints.NotNull;
 
 public class GroupCreationDTO {
     @NotNull(message = "Group name cannot be null")
     private String groupName;
    
-    private String userEmail;
+    private List<String> userEmail;
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/dto/GroupCreationDTO.java]: userEmails with s if you create a list

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/leveluptasks/service/GroupService.java [hunk 6]
```diff
@@ -61,49 +64,39 @@ public Groupe addUserGroupe (Long id , List<String> UserEmail) {
...
-        Groupe groupe = new Groupe();
-        groupe.setName(groupCreationDTO.getGroupName());
-        
-
-        List<String> userEmail = groupCreationDTO.getUserEmail();
-        if (userEmail != null && !userEmail.isEmpty()) {
-            for (String email : userEmail) {
+    public Groupe updateUserToGroupe(Long id , List<String> UserEmail) {
+        System.out.println( id);
+        System.out.println( UserEmail);
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/service/GroupService.java]: remove debug prints

File: src/main/java/com/leveluptasks/service/GroupService.java [hunk 7]
```diff
@@ -61,49 +64,39 @@ public Groupe addUserGroupe (Long id , List<String> UserEmail) {
...
+        Groupe groupe = groupRepository.findById(id).orElse(null);
+        System.out.println( groupe);
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/service/GroupService.java]: remove debug prints

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/leveluptasks/service/GroupService.java [hunk 8]
```diff
@@ -75,8 +72,6 @@ public Groupe updateGroupe(Long id,Groupe groupe) {
     }
 
     public Groupe updateUserToGroupe(Long id , List<String> UserEmail) {
-        System.out.println( id);
-        System.out.println( UserEmail);
         Groupe groupe = groupRepository.findById(id).orElse(null);
         System.out.println( groupe);
```
Comment (Benja-san) [inline src/main/java/com/leveluptasks/service/GroupService.java]: still one

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**